packages/android_intent/android/src/test/java/io/flutter/plugins/androidintent/MethodCallHandlerImplTest.java

```diff
@@ -213,6 +213,30 @@ public void onMethodCall_setsComponentName() {
 
     methodCallHandler.onMethodCall(new MethodCall("launch", args), result);
 
+    verify(result, times(1)).success(null);
+    Intent intent = shadowOf((Application) context).getNextStartedActivity();
+    assertNotNull(intent);
+    assertNotNull(intent.getComponent());
+    assertEquals("foo", intent.getAction());
+    assertEquals(expectedComponent.getPackageName(), intent.getPackage());
```

Comment:
Just to confirm, the method below would then read like this:

``` java
sender.setApplicationContext(context);
Map<String, Object> args = new HashMap<>();
ComponentName expectedComponent =
    new ComponentName("io.flutter.plugins.androidintent", "MainActivity");
args.put("package", expectedComponent.getPackageName());
args.put("componentName", expectedComponent.getClassName());
Result result = mock(Result.class);
ShadowPackageManager shadowPm =
    shadowOf(ApplicationProvider.getApplicationContext().getPackageManager());
shadowPm.addActivityIfNotPresent(expectedComponent);

methodCallHandler.onMethodCall(new MethodCall("launch", args), result);

verify(result, times(1)).success(null);
Intent intent = shadowOf((Application) context).getNextStartedActivity();
assertNotNull(intent);
assertNotNull(intent.getComponent());
assertEquals("foo", intent.getAction());
assertEquals(
    "io.flutter.plugins.androidintent/io.flutter.plugins.androidintent.MainActivity", 
        intent.getComponent().flattenToString());
```

The [flattenToString](https://developer.android.com/reference/android/content/ComponentName#flattenToString()) method is interesting, it creates a distinct string that we can use for the comparison. For the moment, I'd probably defer this to another time.

Reply:
I think one assertion is missing from your above sample, but generally yeah that's what I was thinking.

The benefit is that test readers don't need to jump up to the top to find `expectedComponent`, and readers don't need to know what `flattenToString()` does, or what the structure of a component name looks like, because there's a literal statement in the assertion.

Up to you whether to make that change or not.